src/app/services/form-data.service.ts

```diff
@@ -0,0 +1,13 @@
+import { Injectable } from '@angular/core';
+import { of } from 'rxjs';
+import { FormElementBase } from '../shared/dynamic-form/types/form-element-base';
+import * as FormData from './form-data.json';
+
+@Injectable()
```

Comment:
It's best to specify where the service is provided. By folder structure we can see this is a service provided in the root, so should have: 
```js
@Injectable({
  providedIn: 'root',
})
```

Reply:
Added.